client/src/util/graphQL.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-console */
+import axios from 'axios';
+
+interface GitLabData {
+    data: {
+        currentUser: {
+        id: string;
+        name: string;
+        username: string;
+        email: string;
+        };
+    }
+}
+  
+export default async function fetchGitLabGraphQLQuery(
+  accessToken: string | null,
+  query: string,
+  gitlabUrl: string,
+): Promise<GitLabData | null> {
+  try {
+    const response = await axios.post<GitLabData>(gitlabUrl, { query }, {
```

Comment:
The [apollo client](https://www.apollographql.com/docs/react/) is a really good choice. Is there any reason to use axios for graphql queries?